app/controllers/businessmatching/RegisterServicesController.scala

```diff
@@ -96,34 +96,14 @@ class RegisterServicesController @Inject()(val authConnector: AuthConnector,
               }
             }
           case ValidForm(_, data) =>
-            (for {
-              isPreSubmission <- statusService.isPreSubmission
-              businessMatching <- businessMatchingService.getModel.value
-              savedModel <- updateModel(
-                businessMatching,
-                newModel(businessMatching.activities,
-                  data,
-                  isPreSubmission
-                ),
-                isMsb(data, businessMatching.activities)
-              )
-              _ <- maybeRemoveAccountantForAMLSRegulations(savedModel)
-              _ <- clearRemovedSections(
-                businessMatching.activities.getOrElse(
-                  BusinessMatchingActivities(
-                    Set()
-                  )
-                ).businessActivities,
-                savedModel.businessActivities
-              )
-            } yield savedModel) flatMap { savedActivities =>
+            formatBusinessActivities(data) flatMap { action =>
```

Comment:
We can't decompose this into (isRemoving, savedModel) ?

Reply:
If we can't, I wish we could.

Reply:
Might be worth naming them as val before we pass them into the `val rps` declaration.